src/userdev/java/net/minecraftforge/gradle/userdev/UserDevPlugin.java

```diff
@@ -316,4 +341,25 @@ private NamedDomainObjectContainer<RenameJarInPlace> createReobfExtension(Projec
         project.getExtensions().add("reobf", reobfExtension);
         return reobfExtension;
     }
+
+    protected void configureJarJarTask(Project project) {
+        project.getConfigurations().create(JAR_JAR_DEFAULT_CONFIGURATION_NAME);
+
+        JavaPluginExtension extension = project.getExtensions().getByType(JavaPluginExtension.class);
+        project.getTasks().register(JAR_JAR_TASK_NAME, JarJar.class, jarJar -> {
+            jarJar.setGroup(JAR_JAR_GROUP);
+            jarJar.setDescription("Create a combined JAR of project and selected dependencies");
+            jarJar.getArchiveClassifier().convention("all");
+
+            jarJar.getManifest().inheritFrom(((Jar) project.getTasks().getByName("jar")).getManifest());
+            jarJar.from(extension.getSourceSets().getByName("main").getOutput());
+
+            jarJar.configuration(project.getConfigurations().getByName(JAR_JAR_DEFAULT_CONFIGURATION_NAME));
```

Comment:
I think this would be better if you store the configuration you created just a few lines earlier.